src/commands/generate.ts

```diff
@@ -1,26 +1,260 @@
-import {Args, Flags} from '@oclif/core'
+/* eslint-disable unicorn/no-await-expression-member */
+import {Args, Errors, Flags} from '@oclif/core'
+import chalk from 'chalk'
+import {readFile, rm, writeFile} from 'node:fs/promises'
+import {join, resolve, sep} from 'node:path'
+import validatePkgName from 'validate-npm-package-name'
 
-import CommandBase from './../command-base'
+import {FlaggablePrompt, GeneratorCommand, exec, makeFlags, readPJSON} from '../generator'
+import {compact, uniq, validateBin} from '../util'
 
-export default class Generate extends CommandBase {
+async function fetchGithubUserFromAPI(): Promise<{login: string; name: string} | undefined> {
+  const token = process.env.GITHUB_TOKEN ?? process.env.GH_TOKEN
+  if (!token) return
+
+  const {default: got} = await import('got')
+  const headers = {
+    Accept: 'application/vnd.github.v3+json',
+    Authorization: `Bearer ${token}`,
+  }
+
+  try {
+    const {login, name} = await got('https://api.github.com/user', {headers}).json<{login: string; name: string}>()
+    return {login, name}
+  } catch {}
+}
```

Comment:
I couldn't find where yeoman was using this env, seems it did a while ago but now it does an unauthenticated request, and relies on the user having an email setup in git:

https://github.com/yeoman/generator/blob/71dcba00427d7e382c1c32af68dbb7666ee0269f/src/actions/user.ts#L51-L52

https://github.com/sindresorhus/github-username/blob/c716cf8e920430dc319c44ab7586ab2e72b91467/index.js#L18

Either way works so it should be fine, the precedence seems for getting the username looks like:
1) GH API (if token)
2) git email (if available)
3) pjson.author in template

I'll check on QA 👍🏼